src/net/p2p/server.cpp

```diff
@@ -0,0 +1,91 @@
+#include "server.h"
+
+namespace P2P {
+  void ServerListener::do_accept() {
+    this->acceptor_.async_accept(
+      net::make_strand(this->io_context_),
+      boost::beast::bind_front_handler(
+        &ServerListener::on_accept,
+        shared_from_this()
+      )
+    );
+  }
+
+  void ServerListener::on_accept(boost::system::error_code ec, net::ip::tcp::socket socket) {
+    Utils::logToDebug(Log::P2PServer, __func__, "New connection.");
+    if (ec) {
+      Utils::logToDebug(Log::P2PServer, __func__, "Error accepting connection: " + ec.message());
+      /// TODO: Handle error
+      return;
+    } else {
+      std::make_shared<Session>(std::move(socket), ConnectionType::INBOUND, this->manager_, this->threadPool_)->run();
+    }
+    this->do_accept();
+  }
+
+  void ServerListener::run() {
+    this->do_accept();
+  }
+
+  void ServerListener::stop() {
+    // Cancel is not available under Windows systems
+    #ifndef __MINGW32__
```

Comment:
Why? We are using docker in Windows. Even if we're planning to have a full windows support for the SDK, Visual Studio does not use MINGW.

Reply:
Condition removed and error handling added for .cancel() and .close()